src/H5Tconv.c

```diff
@@ -307,6 +348,23 @@ H5T__conv_order(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata,
                     } /* end if */
                     break;
 
+                case H5T_COMPLEX: {
+                    const H5T_shared_t *src_base_sh = src->shared->parent->shared;
```

Comment:
It would be more compact, though `H5T_cmp` will compare the datatypes as different if they have different byte orders and then there's no information about whether or not that was the only thing different about the datatypes.